CMakeLists.txt

```diff
@@ -28,6 +28,7 @@ option(USE_OPENMP "Build with Openmp support" ON)
 cmake_dependent_option(USE_CUDNN "Build with cudnn support" ON "USE_CUDA" OFF) # one could set CUDNN_ROOT for search path
 cmake_dependent_option(USE_SSE "Build with x86 SSE instruction support" ON "NOT ARM" OFF)
 option(USE_F16C "Build with x86 F16C instruction support" ON) # autodetects support if ON
+option(USE_X86_ARCH         "Build with x86 options for -march" OFF)
```

Comment:
`option` can only be `ON` or `OFF`. I suspect users want to specify string values such as `native` here.

Reply:
Thanks! How about changing to:
`set(USE_X86_ARCH "" CACHE STRING "Build with x86 options for -march")`

Reply:
Yes. But please see my question about default value below.

Reply:
@leezu it actually accepts other values that ON OFF, it's a bit misleading.